lua/neogit/buffers/commit_editor/init.lua

```diff
@@ -40,25 +40,25 @@ function M:open()
     modifiable = true,
     readonly = false,
     autocmds = {
-      ["BufWritePre"] = function()
-        written = true
-      end,
       ["BufUnload"] = function(o)
-        if written then
+        local buf = Buffer.create {
+          name = o.buf,
+        }
```

Comment:
Could you explain what this is doing? Not sure I follow the logic. It's creating a new buffer to check the modified status?

Reply:
So previously we were directly interfacing with the nvim api to manage this `o.buf`, but we already have a built-in buffer class that has a bunch of these management functions.

I'm constructing our internal Buffer class so we can use in-built functionality from our own lib.